R/data_xtabulate.R

```diff
@@ -12,30 +12,30 @@
   }
   # frequency table
   if (is.null(weights)) {
-    if (include_na) {
-      x_table <- tryCatch(table(addNA(x), addNA(by)), error = function(e) NULL)
-    } else {
+    if (remove_na) {
       x_table <- tryCatch(table(x, by), error = function(e) NULL)
+    } else {
+      x_table <- tryCatch(table(addNA(x), addNA(by)), error = function(e) NULL)
```

Comment:
Why do we need `tryCatch()` here? It would be good to add a comment explaining why it's needed so that nobody in future unintentionally removes it.

Reply:
I think because it's a `.default` method, which could accept objects that don't work with `table()`. We then later have

```r
  if (is.null(freq_table)) {
    insight::format_warning(paste0("Can't compute frequency tables for objects of class `", class(x)[1], "`."))
    return(NULL)
  }
```